### PR TITLE
Refactor change language into a linear sequence

### LINKED ISSUE #6504: [8.4.1+nightly] Returning to main menu causes exception claiming missing style
After updating to the latest `8.4.1.25071704+nightly` (before it was the `+nightly` before that, also `8.4.1`), my projects (!) crash when starting a game and then entering the main menu again. I didn't change anything related to styles or the main menu. This also happens to a barely touched new test project.

The exception reads as follows:

```
I'm sorry, but an uncaught exception occurred.

While running game code:
  File "renpy/style.pyx", line 96, in renpy.style.get_style
  File "renpy/style.pyx", line 83, in renpy.style.get_style
Exception: Style 'title' does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "renpy/style.pyx", line 96, in renpy.style.get_style
  File "renpy/style.pyx", line 98, in renpy.style.get_style
Exception: Style 'menu_title' does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tmp/gen3-static/styleclass.pxi", line 621, in renpy.styledata.styleclass.Style.slow_abortable.__get__
  File "renpy/style.pyx", line 493, in renpy.style.StyleCore._get
  File "renpy/style.pyx", line 686, in renpy.style.build_style
  File "renpy/style.pyx", line 142, in renpy.style.get_full_style
  File "renpy/style.pyx", line 98, in renpy.style.get_style
Exception: Style 'main_menu_title' does not exist.

-- Full Traceback ------------------------------------------------------------

Traceback (most recent call last):
  File "renpy/style.pyx", line 96, in renpy.style.get_style
  File "renpy/style.pyx", line 83, in renpy.style.get_style
Exception: Style 'title' does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "renpy/style.pyx", line 96, in renpy.style.get_style
  File "renpy/style.pyx", line 98, in renpy.style.get_style
Exception: Style 'menu_title' does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in script
    python hide:
  File "renpy/ast.py", line 1187, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "renpy/python.py", line 1260, in py_exec_bytecode
    exec(bytecode, globals, locals)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "renpy/common/_layout/screen_main_menu.rpym", line 28, in <module>
    python hide:
  File "renpy/common/_layout/screen_main_menu.rpym", line 35, in _execute_python_hide
    ui.interact()
    ~~~~~~~~~~~^^
  File "renpy/ui.py", line 304, in interact
    rv = renpy.game.interface.interact(roll_forward=roll_forward, **kwargs)
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "renpy/display/core.py", line 2216, in interact
    repeat, rv = self.interact_core(
                 ~~~~~~~~~~~~~~~~~~^
        preloads=preloads,
        ^^^^^^^^^^^^^^^^^^
    ...<4 lines>...
        **kwargs,
        ^^^^^^^^^
    )  # type: ignore
    ^                
  File "renpy/display/core.py", line 2754, in interact_core
    root_widget.visit_all(lambda d: d.per_interact())
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 432, in visit_all
    d.visit_all(callback, seen)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 432, in visit_all
    d.visit_all(callback, seen)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 432, in visit_all
    d.visit_all(callback, seen)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "renpy/display/screen.py", line 508, in visit_all
    self.child.visit_all(callback, seen=None)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 432, in visit_all
    d.visit_all(callback, seen)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 432, in visit_all
    d.visit_all(callback, seen)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^
  File "renpy/display/displayable.py", line 434, in visit_all
    callback(self)
    ~~~~~~~~^^^^^^
  File "renpy/display/core.py", line 2754, in <lambda>
    root_widget.visit_all(lambda d: d.per_interact())
                                    ~~~~~~~~~~~~~~^^ 
  File "renpy/text/text.py", line 2354, in per_interact
    if self.style.slow_abortable:
       ^^^^^^^^^^^^^^^^^^^^^^^^^ 
  File "tmp/gen3-static/styleclass.pxi", line 621, in renpy.styledata.styleclass.Style.slow_abortable.__get__
  File "renpy/style.pyx", line 493, in renpy.style.StyleCore._get
  File "renpy/style.pyx", line 686, in renpy.style.build_style
  File "renpy/style.pyx", line 142, in renpy.style.get_full_style
  File "renpy/style.pyx", line 98, in renpy.style.get_style
Exception: Style 'main_menu_title' does not exist.

Linux-5.15.0-143-generic-x86_64-with-glibc2.31 x86_64
Ren'Py 8.4.1.25071704+nightly
Thu Jul 17 17:25:33 2025
```

### PR DESCRIPTION
**Pull request by mal**
Seeks to resolve #6504.

Currently a first pass at getting us back to a stable base for `change_language`. Good chance this is not it's "final form" but this presentation should make it easier to view and reason about the code path as a whole, along with the various subpaths it needs to support. The hope is that it sets us in good stead for continuing this refactor now or in future should we decide it needs further clean up or optimisation.

Currently the old and new change language order code has been rolled into the main function, this seemed worth doing while seeking to understand and review the whole pipeline, but may make sense to factor out again later.

I've done some cursory testing already involving reloads, end-of-game rollover (back to main menu), save/load cycles, language changes, gui preference changes, and various combinations of these together. So far everything appears to function as it should, but with there being so many edge cases it's possible there are still some out there so I'll be continuing to test.

### CODE REVIEW

File: renpy/common/00gui.rpy [hunk 1]
```diff
@@ -125,17 +128,12 @@ init -1150 python in gui:
...
-    def rebuild():
-        """
-        :doc: gui
-
-        Rebuilds the GUI.
-
-        Note: This is a very slow function.
-        """
+        for i in config.translate_clean_stores:
+            renpy.python.clean_store_backup.backup_one("store." + i)
```
Comment (renpytom) [inline renpy/common/00gui.rpy:132]: This was there previously, but feels kind of unprincipled to me. Specifically, it's not clear why we want to run this here, but not when the language is changed.

Comment (mal) [reply renpy/common/00gui.rpy:132]: Yeah, I'm not a fan either, but best I can determine it is needed to retain and carry updated values through the main menu/end-of-game cycle. Without it, values in the `gui` store reset to whatever the were when the game originally launched, even if they've been updated since, and it's possible to end up in a situation where the values visible in `gui` do not match the styles being seen (even though the styles are correct the `gui` values are lagging behind).
